src/interfaces/chain.cpp

```diff
@@ -326,7 +326,11 @@ class ChainImpl : public Chain
     CFeeRate relayIncrementalFee() override { return ::incrementalRelayFee; }
     CFeeRate relayDustFee() override { return ::dustRelayFee; }
     CAmount maxTxFee() override { return ::maxTxFee; }
-    bool getPruneMode() override { return ::fPruneMode; }
+    bool havePruned() override
```

Comment:
Move to the chain lock?

Reply:
Why?

Reply:
Sorry for the brief reply, but see here:

https://github.com/bitcoin/bitcoin/blob/fad84519391a05f2ad42381cca9135323b709657/src/interfaces/chain.h#L69-L73

The chain lock is deprecated and will be removed in the future.